ai-service/src/ai_service/embeddings/tests/test_transformer.py

```
@@ -0,0 +1,110 @@
+from ai_service import errors
+from ai_service.embeddings import transformer
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+class TestDeviceDetection:
+    """Test device detection functionality."""
+
+    @pytest.mark.parametrize(
+        "cuda_available,mps_available,expected_device,test_name",
```

Ruff (0.12.2)

Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)